src/main/java/com/google/sps/data/User.java

```diff
@@ -0,0 +1,205 @@
+      this.bookmarked_events = new ArrayList<Long>();
+      this.added_to_calendar_events = new ArrayList<Long>();
+      this.current_challenge_id = 0L;
+      this.challenge_statuses = new ArrayList<Integer>();
+      this.entity_key = null;
+   }
+
+
+
+  // @Erick May need to change the following methods if structure of challenge statuses or id changes
+  public Long getCurrentChallenge() {
+      return this.current_challenge_id;
+  }
+
+  public void setCurrentChallenge(Long chal_id) {
+      this.current_challenge_id = chal_id;
+  }
+
+  public ArrayList<Integer> getChallengeStatuses() {
+      return (ArrayList) this.challenge_statuses;
+  }
+
+  // challenge_statuses param should not be null
+  public void setChallengeStatuses(ArrayList<Integer> challenge_statuses) {
+      this.challenge_statuses = (ArrayList) challenge_statuses.clone();
+  }
+
+  public ArrayList<Long> getCreatedEvents() {
+      return (ArrayList) this.created_events;
+  }
+
+  // created_events param should not be null
+  public void setCreatedEvents(ArrayList<Long> created_events) {
+      this.created_events = (ArrayList) created_events.clone();
+  }
+
+  public ArrayList<Long> getBookmarkedEvents() {
+      return (ArrayList) this.bookmarked_events;
+  }
+
+  // bookmarked_events param should not be null
+  public void setBookmarkedEvents(ArrayList<Long> bookmarked_events) {
+      this.bookmarked_events = (ArrayList) bookmarked_events.clone();
+  }
+
+  public ArrayList<Long> getAddedToCalendarEvents() {
+      return (ArrayList) this.added_to_calendar_events;
+  }
+
+  // added_to_calendar_events param should not be null
+  public void setAddedToCalendarEvents(ArrayList<Long> added_to_calendar_events) {
+      this.added_to_calendar_events = (ArrayList) added_to_calendar_events.clone();
+  }
+ 
+
+  public static User convertEntitytoUser(Entity entity, String userId) {
+    Key entityKey = entity.getKey();
+    String nickname = (String) entity.getProperty(NICKNAME);
+    Long currentChallengeId = (Long) entity.getProperty(CURRENT_CHALLENGE);
+    ArrayList<Long> createdEvents =(ArrayList<Long>) entity.getProperty(CREATED_EVENTS);
+    ArrayList<Long> bookmarkedEvents = (ArrayList<Long>) entity.getProperty(BOOKMARKED_EVENTS);
+    ArrayList<Long> addedEvents = (ArrayList<Long>) entity.getProperty(ADDED_TO_CALENDAR_EVENTS);
+    ArrayList<Integer> challengeStatuses = (ArrayList<Integer>) entity.getProperty(CHALLENGE_STATUSES);
+
+    User user = new User.Builder(userId)
+        .setNickname(nickname)
+        .setEntityKey(entityKey)
+        .setCreatedEvents(createdEvents)
+        .setBookmarkedEvents(bookmarkedEvents)
+        .setAddedToCalendarEvents(addedEvents)
+        .setCurrentChallengeId(currentChallengeId)
+        .setChallengeStatuses(challengeStatuses)
+        .build();
+
+    return user;
+  }
+  
+  public Entity toEntity() {
+      Entity userEntity;
+      if (this.entity_key == null) userEntity = new Entity(DATA_TYPE);
```

Comment:
you need to update entity_key here.

If you call toEntity() multiple time, we might end up creating two different Entities with different keys right?